base/file.jl

```diff
@@ -356,12 +339,33 @@ is an open file object for this path.
 mktemp(parent)
 
 """
-    mktempdir(parent=tempdir())
+    mktempdir(parent=tempdir(); prefix="$temp_prefix")
 
 Create a temporary directory in the `parent` directory and return its path.
-If `parent` does not exist, throw an error.
+If `parent` does not exist, throw an error. An optional `prefix` to the directory name can
+be provided.
 """
-mktempdir(parent)
+function mktempdir(parent=tempdir(); prefix="$temp_prefix")
+    template = prefix*"XXXXXX"
+    tpath = joinpath(parent, template)
+
+    req = Libc.malloc(_sizeof_uv_fs)
+    try
+        ret = ccall(:uv_fs_mkdtemp, Int32,
+            (Ptr{Void}, Ptr{Void}, Cstring, Ptr{Void}),
```

Comment:
I followed the same indentation as `readlink`, although my personal preference would be to align to ccall.